app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarMenuController.kt

```diff
@@ -184,7 +187,7 @@ class DefaultBrowserToolbarMenuController(
                 val directions = NavGraphDirections.actionGlobalShareFragment(
                     data = arrayOf(
                         ShareData(
-                            url = currentSession?.url,
+                            url = getProperUrl(currentSession),
```

Comment:
nit: I would call this `getShareableUrl` or `ensureUrlShareable`.

Reply:
I was thinking maybe using this method also for another issue, with the bookmarks (https://github.com/mozilla-mobile/fenix/issues/15413). That is why I wanted to leave it more generic.

Reply:
ok :). We you tackle that we could also uplift this function to A-C. It seems it could be useful it many places and we could add an extension function `TabSessionState` for example.